### PR TITLE
fix: replace extension with .ngfactory.ts

### PR DESCRIPTION
**Pull request by olivercs**
Fix to issue #4238

### CODE REVIEW

**Comment by Meligy** [comment]
@olivercs Please check the build error.

Also can this have an e2e test?

Thanks.

**Comment by olivercs** [comment]
@Meligy not sure how to do it.

File: packages/@ngtools/webpack/src/loader.ts [hunk 1]
```diff
@@ -163,7 +163,7 @@ function _replaceBootstrap(plugin: AotPlugin, refactor: TypeScriptFileRefactor)
   const genDir = path.normalize(plugin.genDir);
   const dirName = path.normalize(path.dirname(refactor.fileName));
   const entryModule = plugin.entryModule;
-  const entryModuleFileName = path.normalize(entryModule.path + '.ngfactory');
+  const entryModuleFileName = path.normalize(entryModule.path.replace(/\.ts$/, '.ngfactory.ts'));
```
Comment (hansl) [inline packages/@ngtools/webpack/src/loader.ts]: This has a bug where the path does NOT include `.ts` at the end.

Comment (olivercs) [inline packages/@ngtools/webpack/src/loader.ts]: @hansl, is safe to append the string '.ngfactory.ts' in that particular case?
```
 if( extension === '.ts' ) {
 	entryModuleFileName = path.normalize(entryModule.path.replace(/\.ts$/, '.ngfactory.ts'));
 }else if(extension === '') {
 	entryModuleFileName = path.normalize(entryModule.path + '.ngfactory.ts');
 }
```

File: packages/@ngtools/webpack/src/loader.ts [hunk 2]
```diff
@@ -212,7 +212,8 @@ function _replaceBootstrap(plugin: AotPlugin, refactor: TypeScriptFileRefactor)
     });
 
   refactor.insertImport('platformBrowser', '@angular/platform-browser');
-  refactor.insertImport(entryModule.className + 'NgFactory', ngFactoryPath);
+  refactor.insertImport(entryModule.className + 'NgFactory', ngFactoryPath + '.ngfactory.ts');
```
Comment (hansl) [inline packages/@ngtools/webpack/src/loader.ts]:
```typescript
import A from 'some/path/file.ngfactory.ts'
```

is wrong. You should not include the `ts` itself in the import path.

**Comment by Brocco** [comment]
Corresponding issue has been closed as obsolete, closing this PR.

**Pull request closed without merging**